include/libxnvme_libconf.h

```diff
@@ -9,7 +9,6 @@
 #ifndef __LIBXNVME_LIBCONF_H
```

Comment:
Should this include guard be removed as well?

Reply:
Unfortunately not, due to:
```
The header-guards / ifdef and C++ wrapping is now provided in a single
    place, that is, in the library-entrypoint header "libxnvme.h". All
    public API headers now drop these guards as they hinder the
    header-composability introduced by "headers include no other headers".

    An exception to this are the following files:

    * libxnvme_libconf.h
    * libxnvme_nvm.h
    * libxnvme_spec_fs.h
    * libxnvme_znd.h

    Since fio consumes these, then it would break upstream fio. Thus,
    punting the drop of header guards in the above files until upstream fio
    is changed to consume xNVMe v0.7.0.

    All the tools, tests, and examples currently consuming the xNVMe headers
    are changed to use the library entry-point "libxnvme.h". This is
    required to avoid duplicate definitions.

    The changes to library-consumption could also have been done by
    individual header-selection, however, using the entry-point verifies
    that the entry-point is correct. E.g. "dog-fooding" the interface.

    Signed-off-by: Simon A. F. Lund <simon.lund@samsung.com>
```